addons/web_editor/static/lib/odoo-editor/src/OdooEditor.js

```diff
@@ -1425,12 +1444,32 @@ export class OdooEditor extends EventTarget {
             }
         }
         this.updateColorpickerLabels();
+        const mapNodeNodeNameToTypeName = {
+            'H1': 'Heading 1',
+            'H2': 'Heading 2',
+            'H3': 'Heading 3',
+            'H4': 'Heading 4',
+            'H5': 'Heading 5',
+            'H6': 'Heading 6',
+            'PRE': 'Code',
+            'BLOCKQUOTE': 'Blockquote',
+        };
+        const possibleTags = new Set(Object.keys(mapNodeNodeNameToTypeName));
+        const firstNode = [sel.anchorNode, ...ancestors(sel.anchorNode)].find(node =>
+            possibleTags.has(node.nodeName),
+        );
+        const typeName = firstNode ? mapNodeNodeNameToTypeName[firstNode.nodeName] : 'Normal';
```

Comment:
- How are these translated?
- mapNodeNode? Not sure to understand the variable name :grimacing: